### DISCUSSION TITLE
ESM Import File extension substitution

### DISCUSSION THREAD

**Discussion by xseman** [denoland/deno#21203] [Q&A] [answered]
I'm currently exploring ESM modules and am curious about whether Deno supports the file extension substitution mentioned in TypeScript [docs](https://www.typescriptlang.org/docs/handbook/modules/reference.html#file-extension-substitution).

```
// foo.ts
export const foo = 'foo';

// bar.ts
import { foo } from './foo.js';
```

```sh
deno run bar.ts
# error: Module not found "file:///tmp/tmp.lVAcziIwpe/foo.js".
#    at file:///tmp/tmp.lVAcziIwpe/bar.ts:1:21
```

**Comment by bartlomieju** [comment]
No, this is currently not supported.

**Comment by dsherret** [comment, marked as answer]
This is now supported via `--unstable-sloppy-imports` in Deno 1.39: https://deno.com/blog/v1.39#sloppy-imports
